### DISCUSSION TITLE
Check if customer is logged in

### DISCUSSION THREAD

**Discussion by hallleron** [pimcore/pimcore#10099] [Q&A / Support]
Hey developers,

is there a way to check if a customer is logged in the frontend (I mean a customer from the **CustomerManagementFrameworkBundle** in the demo shop installation). I need to programmatically check, if a customer is logged in or not. I didn't find a matching function in the CustomerProviderInterface.

Thank you very much for your help!

**Comment by aarongerig** [comment]
The easiest would be to check for the `ROLE_USER` role. Any authenticated user should have at least that role, which means you can check for it in order to know, if the user is logged in or not.

**PHP Controller**
```php
if ($this->isGranted('ROLE_USER')) {
    // User is logged in
} else {
    // Not logged in
}
```

**Twig**
```twig
{% if is_granted('ROLE_USER') %}
    {# User is logged in #}
{% else %}
    {# Not logged in #}
{% endif %}
```

**Reply by hallleron** [reply]
Does this also work for customers? And I need this in an Event Listener Class, not a controller. So which class do I need in order to call the function "isGranted"? Thank you in advance =)

**Reply by aarongerig** [reply]
Yes, that should also work for customers. In the end you are using the Symfony security component to log in your users/customers.
In a service (other than controller), you can inject the security helper service and call `isGranted()` on that.

**EventListener.php**
```php
use Symfony\Component\EventDispatcher\EventSubscriberInterface;
use Symfony\Component\Security\Core\Security;

class EventListener implements EventSubscriberInterface
{
    private Security $security;

    public function __construct(Security $security)
    {
        $this->security = $security;
    }
    
    ...
    
    public function isUserLoggedIn(): bool
    {
        return $this->security->isGranted('ROLE_USER');
    }
}
```

**Reply by hallleron** [reply]
Okay thank you very much! That does seem to work.
